### ISSUE TITLE
The cake script is failed with NuGet.Protocol 5.8.0

### ISSUE THREAD

**Issue by AnastasiaKubova** [NuGet/Home#10245]
## Details about Problem

NuGet product used (NuGet.exe):

NuGet - latest version

## Detailed repro steps so we can see the same problem

After the release of the latest version of `NuGet.Protocol` our cake scripts began to crash with the following error:

```
Error: NuGet.Resolver.NuGetResolverConstraintException: Unable to resolve dependency 'NuGet.Packaging'. Source(s) used: '/Users/runner/work/1/s/tools/Addins', '/Users/runner/.nuget/packages/', 'nuget.org'.
  at NuGet.Resolver.PackageResolver.Resolve (NuGet.Resolver.PackageResolverContext context, System.Threading.CancellationToken token) [0x00544] in <0238e4cd771a452c873dc6fe9e575776>:0 
  at Cake.NuGet.Install.NuGetPackageInstaller.Install (Cake.Core.Packaging.PackageReference package, Cake.Core.Packaging.PackageType type, Cake.Core.IO.DirectoryPath path) [0x00194] in <51706cc3401f4abd8da7efce44e907ce>:0 
  at Cake.Core.Scripting.ScriptProcessor.InstallAddins (System.Collections.Generic.IReadOnlyCollection`1[T] addins, Cake.Core.IO.DirectoryPath installPath) [0x00095] in <352990cac81e4748b475ea51792a9c60>:0 
  at Cake.Core.Scripting.ScriptRunner.Run (Cake.Core.Scripting.IScriptHost host, Cake.Core.IO.FilePath scriptPath, System.Collections.Generic.IDictionary`2[TKey,TValue] arguments) [0x00144] in <352990cac81e4748b475ea51792a9c60>:0 
  at Cake.Commands.BuildCommand.Execute (Cake.CakeOptions options) [0x0003f] in <f7ab677035254e96ac028557e3c3278e>:0 
  at Cake.CakeApplication.Run (Cake.CakeOptions options) [0x00015] in <f7ab677035254e96ac028557e3c3278e>:0 
  at Cake.Program.Main () [0x000d1] in <f7ab677035254e96ac028557e3c3278e>:0 
```

We used this dependency as follows:
`#addin nuget:?package=NuGet.Protocol&loaddependencies=true`

To avoid this we had to explicitly specify the previous version of  `NuGet.Protocol`:
`#addin nuget:?package=NuGet.Protocol&version=5.7.0&loaddependencies=true`

### Sample Project

https://github.com/microsoft/appcenter-sdk-unity/pull/477

**Comment by paulwoodland** [comment]
I'm having a similar problem, the issue seems to be that the "Nuget.Packging" package for 5.8 hasn't been released, it is still only 5.7 available, or the previews of 5.8.

**Comment by zivkan** [comment]
Although this issue was created first, I'm closing it as a duplicate of #10246 as it has a more precise title.
